src/main/java/com/atlauncher/gui/tabs/VanillaPacksTab.java

```diff
@@ -412,523 +251,292 @@ public void actionPerformed(ActionEvent e) {
-            } catch (InvalidMinecraftVersion ignored) {
-                createServerButton.setVisible(false);
-            }
+                // ensures that the dropdown is at least 200 px wide
+                loaderVersionLength = max(200, loaderVersionLength);
 
-            if (!nameFieldDirty) {
-                nameField.setText(defaultValue);
-            }
+                // ensures that there is a maximum width of 400 px to prevent overflow
+                loaderVersionLength = min(400, loaderVersionLength);
+                loaderVersionsDropDown.setPreferredSize(new Dimension(loaderVersionLength, 23));
 
-            if (!descriptionFieldDirty) {
-                descriptionField.setText(defaultValue);
+                selectionDisposable = viewModel.selectedLoaderVersion().subscribe((it) -> {
+                    it.ifPresent(loaderVersion -> loaderVersionsDropDown.setSelectedIndex(loaderVersions.indexOf(loaderVersion)));
+                });
             }
-
-            loaderTypeFabricRadioButton.setVisible(
-                    !ConfigManager.getConfigItem("loaders.fabric.disabledMinecraftVersions", new ArrayList<String>())
-                            .contains(newSelectedMinecraftVersion));
-            loaderTypeForgeRadioButton.setVisible(
-                    !ConfigManager.getConfigItem("loaders.forge.disabledMinecraftVersions", new ArrayList<String>())
-                            .contains(newSelectedMinecraftVersion));
-            loaderTypeLegacyFabricRadioButton.setVisible(
-                    !ConfigManager
-                            .getConfigItem("loaders.legacyfabric.disabledMinecraftVersions", new ArrayList<String>())
-                            .contains(newSelectedMinecraftVersion));
-            loaderTypeQuiltRadioButton.setVisible(
-                    !ConfigManager.getConfigItem("loaders.quilt.disabledMinecraftVersions", new ArrayList<String>())
-                            .contains(newSelectedMinecraftVersion));
-
-            // refresh the loader versions if we have one selected
-            LoaderType selectedLoaderType = getSelectedLoader();
-            if (selectedLoaderType != null) {
-                selectedLoaderTypeChanged(selectedLoaderType);
+        });
+        viewModel.loaderLoading().subscribe((it) -> {
+            loaderVersionsDropDown.removeAllItems();
+            if (it) {
+                loaderVersionsDropDown.addItem(new ComboItem<>(null, GetText.tr("Getting Loader Versions")));
+            } else {
+                setEmpty();
             }
-        }
+        });
+        mainPanel.add(loaderVersionsDropDown, gbc);
+        add(mainPanel, BorderLayout.CENTER);
     }
 
-    private LoaderType getSelectedLoader() {
-        if (loaderTypeFabricRadioButton.isSelected()) {
-            return LoaderType.FABRIC;
-        }
+    private void setEmpty() {
+        loaderVersionsDropDown.addItem(new ComboItem<>(null, GetText.tr("Select Loader First")));
```

Comment:
This doesn't seem to get called anywhere, changing from a loader to None, the combobox is just empty (Current on the left, this PR on the right).

![image](https://user-images.githubusercontent.com/1723537/234814681-7d2686d6-25b0-445c-90a0-b65c4e2151e4.png)

It flashes up for a second but then goes back to being empty

Reply:
[7dc9e7d](https://github.com/ATLauncher/ATLauncher/pull/743/commits/7dc9e7d7897fe02e3429231522f6d698db9f97bb)